File: app/src/main/java/com/jeeps/gamecollector/comparators/GameByHoursStoryComparator.java
package com.jeeps.gamecollector.comparators;

import com.jeeps.gamecollector.remaster.data.model.data.games.Game;
import com.jeeps.gamecollector.remaster.data.model.data.games.GameHoursStats;

import java.util.Comparator;

public class GameByHoursStoryComparator implements Comparator<Game> {
    private final boolean desc;

    public GameByHoursStoryComparator(boolean desc) {
        this.desc = desc;
    }

    public GameByHoursStoryComparator() {
        this.desc = false;
    }

    @Override
    public int compare(Game game1, Game game2) {
        game1.getGameHoursStats();
        GameHoursStats gameHoursStats1 = game1.getGameHoursStats();

        game2.getGameHoursStats();
        GameHoursStats gameHoursStats2 = game2.getGameHoursStats();
        if (!desc)
            return Double.compare(
                    gameHoursStats1.getGameplayMain(),
                    gameHoursStats2.getGameplayMain());
        return Double.compare(
                gameHoursStats1.getGameplayMain(),
                gameHoursStats2.getGameplayMain()) * -1;
    }
}

File: app/src/main/java/com/jeeps/gamecollector/remaster/utils/comparators/GameByHoursStoryComparator.kt
package com.jeeps.gamecollector.remaster.utils.comparators

import com.jeeps.gamecollector.remaster.data.model.data.games.Game

class GameByHoursStoryComparator : Comparator<Game> {
    private val desc: Boolean

    constructor(desc: Boolean) {
        this.desc = desc
    }

    constructor() {
        desc = false
    }

    override fun compare(game1: Game, game2: Game): Int {
        val (hours1) = game1.gameHoursStats
        val (hours2) = game2.gameHoursStats
        return if (!desc)
            hours1.compareTo(hours2)
        else
            hours1.compareTo(hours2) * -1
    }
}